core/src/main/java/org/pentaho/di/core/database/BaseDatabaseMeta.java

```diff
@@ -1438,12 +1438,16 @@ public Properties getConnectionPoolingProperties() {
   @Override
   public void setConnectionPoolingProperties( Properties properties ) {
     // Clear our the previous set of pool parameters
+    ArrayList<String> itemsToRemove = new ArrayList<>();
     for ( Iterator<Object> iter = attributes.keySet().iterator(); iter.hasNext(); ) {
       String key = (String) iter.next();
       if ( key.startsWith( ATTRIBUTE_POOLING_PARAMETER_PREFIX ) ) {
-        attributes.remove( key );
+        itemsToRemove.add( key );
       }
     }
+    for ( String item : itemsToRemove ) {
```

Comment:
Unclear to me why this would be done in a separate loop. This will be much more time-consuming. Suggest this would be a performance regression. 

Suggest instead iterating over the EntrySet of the attributes and calling the remove method of the iterator. There is *no* reason to do things this way other than misunderstanding how to deal with iterators over collections. It should be something like this:

Map.Entry<Object, Object> anEntry;
for ( Iterator<Map.Entry<Object, Object>> iter = attributes.entrySet().iterator(); iter.hasNext(); ) {
  anEntry = iter.next();
  if ( ( (String)anEntry.getKey() ).startsWith( ATTRIBUTE_POOLING_PARAMETER_PREFIX )) {
    iter.remove(); // Removes it from the collection immediately and correctly
  }
)

Reply:
The initial problem was related with list modification while iteration. In fact using iterator to remove directly the object is the best approach :) 
I am going to do this and then send a new PR